### DISCUSSION TITLE
How to get scaffold labels in molecular datasets?

### DISCUSSION THREAD

**Discussion by wzhang2022** [snap-stanford/ogb#159] [General]
In the `ogbg-molhiv` and `ogbg-molpcba` datasets, the train/validation/test split is done via scaffold splitting rather than random. I haven't worked with molecular data before, but my understanding is that since we split based on the structural frameworks of the molecules, we aren't guaranteed i.i.d. data, and we get distribution shift between train/test/evaluation. If I'm interested in getting extracting features of the molecule that are invariant to which scaffold the molecule is in, I might want to know the scaffold labels. Does OGB support this feature?

**Comment by weihua916** [comment]
Hi! You can get scaffold by

```python
from rdkit import Chem
from rdkit.Chem.Scaffolds import MurckoScaffold

def generate_scaffold(smiles, include_chirality=False):
    """return scaffold string of target molecule"""
    mol = Chem.MolFromSmiles(smiles)
    scaffold = MurckoScaffold\
        .MurckoScaffoldSmiles(mol=mol, includeChirality=include_chirality)
    return scaffold
```

**Reply by wzhang2022** [reply]
The [link](https://github.com/snap-stanford/ogb/blob/master/examples/graphproppred/mol/smiles2graph.py) for the script that converts the smiles string to graph objects seems to be broken. How would I get the smiles string for the `ogbg-molhiv` dataset? It doesn't seem to be provided in any of the dataloaders given by OGB.

**Reply by weihua916** [reply]
Sorry for that. The link has been updated: https://github.com/snap-stanford/ogb/blob/master/ogb/utils/mol.py
For SMILES, please see https://github.com/snap-stanford/ogb/issues/90
